controllers/windowsmachine_controller.go

```diff
@@ -120,8 +120,8 @@ func (r *WindowsMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 		// process delete event
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			// for Windows machines only
-			return isWindowsMachine(e.Object.GetLabels())
+			// for Windows machines only without ignore label
+			return isWindowsMachine(e.Object.GetLabels()) && !ignoreLabel(e.Object.GetLabels())
```

Comment:
any reason not to call `r.isValidMachine(e.Object)` instead of introducing `ignoreLabel()`?

Reply:
But the first thing that is checked in `r.isValidMachine(e.Object)` is the label. How are the other validations being hit?

Reply:
True, and it sounds like a hack with a side effect. It was counter-intuitive for me to add a statement to validate a deleted machine. So, I took the label logic apart. 

In the same sense, a machine could be valid with or without the ignore label, WMCO just chooses to ignore it. Hence, I see it as two separate things.

Reply:
What this really calls for is just updating the doc for `isValidMachine()` and not introduce more code. But I won't bikeshed on this.

Reply:
Addressed.